host/sampi/state.go

```diff
@@ -41,12 +47,19 @@ func (s *State) Begin() {
 }
 
 func (s *State) Commit() map[string]host.Host {
-	defer s.Unlock()
+	l := s.logger.New("fn", "Commit")
+	defer func() {
+		l.Debug("unlocking state")
+		s.Unlock()
+	}()
+	l.Debug("checking if hosts have been modified")
 	if !s.nextModified {
+		l.Debug("dropping since hosts where not modified", "at", "nextModified")
 		s.next = nil
 		return s.Get()
 	}
 	// copy hosts that were not modified to next
+	l.Debug("hosts where modified", "at", "next")
```

Comment:
`s/where/were/` but I'd like to see a reduction in logging for this method, as it's a hot-path method that should not be doing IO while the mutex is locked.